etna/collections/models.py

```diff
@@ -309,3 +314,122 @@ def record(self):
         FieldPanel("teaser_image"),
         FieldPanel("description"),
     ]
+
+
+class PageTopic(Orderable):
+    """
+    This model allows any page type to be associated with one or more topics
+    in a way that retains the order of topics selected.
+
+    The ``sort_order`` field value from ``Orderable`` can be used to pull out
+    the 'first' topic to treat as the 'primary topic' for a page, and can also
+    used to prioritise items for a list of 'pages related to a topic'.
+
+    Just add `InlinePanel("page_topics")` to a page type's panel
+    configuration to use it!
+    """
+
+    page = ParentalKey(Page, on_delete=models.CASCADE, related_name="page_topics")
+    topic = models.ForeignKey(
+        TopicExplorerPage,
+        verbose_name=_("topic"),
+        related_name="topic_pages",
+        on_delete=models.SET_NULL,
```

Comment:
Suggestion: I think we would want this to be `models.CASCADE` instead, right? If the `topic` is deleted then this related model should also be, there's no information left to reference.

Reply:
I will give that a go. I could be misremembering, but I thought Wagtail didn't like `models.CASCADE` in inline models, because the content technically belongs to the page, and is in revisions etc.

Reply:
Updated in [072711c](https://github.com/nationalarchives/ds-wagtail/pull/604/commits/072711cb4bc5d40088d777561e54fc2d448a9576)